### PR TITLE
bitbucketcloud: add `apiURL` option

### LINKED ISSUE #7990: Bitbucket Cloud external service doesn't respect custom URLs
Reported by https://app.hubspot.com/contacts/2762526/company/557475593

>there’s already a field for ‘url’ [for the Bitbucket Cloud external service] but entering a value there doesn’t seem to change the URL of the API calls from Sourcegraph to Bitbucket.

### PR DESCRIPTION
**Pull request by unknwon**
This PR adds a `apiURL` external config option to support custom API endpoint. Unit tests are also updated to make sure we respect the config option in all places.

Manually tested to prevent regressions.

Fixes #7990.

### CODE REVIEW

File: cmd/repo-updater/repos/bitbucketcloud_test.go [hunk 1]
```diff
@@ -56,15 +57,16 @@ func TestBitbucketCloudSource_ListRepos(t *testing.T) {
...
+				"staging.bitbucket.org/Unknwon/scripts",
+				"staging.bitbucket.org/Unknwon/wxvote",
+				"staging.bitbucket.org/sglocal/mux",
+				"staging.bitbucket.org/sglocal/go-langserver",
+				"staging.bitbucket.org/sglocal/python-langserver",
 			}),
 			conf: &schema.BitbucketCloudConnection{
-				Url:         "https://bitbucket.org",
+				Url:         "https://staging.bitbucket.org",
+				ApiURL:      "https://api-staging.bitbucket.org",
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/bitbucketcloud_test.go]: these are integration / replay tests. If I am not mistaken you are breaking them here if we actually try to record?

Comment (unknwon) [reply cmd/repo-updater/repos/bitbucketcloud_test.go]: You're right about this. I first tried to set up a proxy to act like they're valid URLs but the proxy also needs to modify the response content from real URLs (api.bitbucket.org -> api-staging.bitbucket.org) so I stopped this approach. What's the best practice here?

Comment (unknwon) [reply cmd/repo-updater/repos/bitbucketcloud_test.go]: The main intention here is to make sure there is no more hard-coded call to "https://api.bitbucket.org", I verified it by hand and it looks like a one-time off effort.

**Comment by unknwon** [comment]
I reverted changes to golden files, waiting CI.

**Pull request merged**